execution_engine_testing/tests/src/test/regression/gov_42.rs

```diff
@@ -0,0 +1,211 @@
+use casper_engine_test_support::{
+    DeployItemBuilder, ExecuteRequestBuilder, InMemoryWasmTestBuilder, DEFAULT_ACCOUNT_ADDR,
+    DEFAULT_PAYMENT, PRODUCTION_RUN_GENESIS_REQUEST,
+};
+use casper_execution_engine::core::engine_state::MAX_PAYMENT;
+use casper_types::{runtime_args, Gas, RuntimeArgs};
+use num_traits::Zero;
+
+use crate::{
+    test::regression::test_utils::{make_gas_counter_overflow, make_module_without_memory_section},
+    wasm_utils,
+};
+
+const ARG_AMOUNT: &str = "amount";
+
+struct TestCase<'a> {
+    pub name: &'a str,
+    pub input_wasm_bytes: &'a [u8],
+    pub expected_error: &'a str,
+}
+
+impl<'a> TestCase<'a> {
+    fn new(name: &'a str, input_wasm_bytes: &'a [u8], expected_error: &'a str) -> Self {
+        Self {
+            name,
+            input_wasm_bytes,
+            expected_error,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+enum ExecutionPhase {
+    Payment,
+    Session,
+}
+
+fn run_test_case(
+    TestCase {
+        name,
+        input_wasm_bytes,
+        expected_error,
+    }: &TestCase,
+    execution_phase: ExecutionPhase,
+) {
+    println!("Running test case '{}' in {:?}", name, execution_phase);
```

Comment:
Left in by mistake?

Reply:
No, that's actually intentional. Thanks to this it's easier to see which test scenario failed.
An alternative would be to split the `should_charge_session_with_incorrect_wasm_files()` into dedicated tests, but this will introduce some copy&pasted boilerplate.

Reply:
I'd prefer the boiler plate over the println

Reply:
> I'd prefer the boiler plate over the println

True, but in this specific case we'll create 10 separate tests to cover each "bad" wasm in both `Payment` and `Session`. See [here](https://github.com/rafal-ch/casper-node/compare/gov_42_zero_cost_deploys_with_bad_wasm_bytes...rafal-ch:refactor_tests_1?expand=1).

@Fraser999 The idea with `scopeguard` is neat, but we can also use the `test-case` I mentioned some time ago. Please look [here](https://github.com/rafal-ch/casper-node/compare/gov_42_zero_cost_deploys_with_bad_wasm_bytes...rafal-ch:refactor_tests_2?expand=1) and especially [here](https://github.com/rafal-ch/casper-node/compare/gov_42_zero_cost_deploys_with_bad_wasm_bytes...rafal-ch:refactor_tests_2?expand=1#diff-08b5ee8febf8ac6b81f2eaf6c72b213f0d6a25c200bb89611998807eb21fb642R59-R68). to see how well it allows to organize the repetitive tests.

Reply:
As agreed, the solution with splitting the test function manually was selected to keep consistency with how we approach similar cases in other places of the system. Fix [here](https://github.com/casper-network/casper-node/pull/3085/commits/f36a87b7c2a895c1e5251d6eb5a5cc3fabf320f0).